core/src/avm1/globals/error.rs

```diff
@@ -0,0 +1,66 @@
+//! Error object
+
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::property::Attribute::*;
+use crate::avm1::{Object, ScriptObject, TObject, UpdateContext, Value};
+use gc_arena::MutationContext;
+
+pub fn constructor<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let message = match args.get(0).cloned() {
+        Some(Value::String(s)) => s,
+        Some(v) => v.coerce_to_string(activation, context)?.to_string(),
+        _ => String::new(),
+    };
+
+    this.set("message", message.into(), activation, context)?;
+
+    // Default name for the object is "Error", which isn't provided by constructor
+    this.set("name", "Error".into(), activation, context)?;
+
+    Ok(Value::Undefined)
+}
+
+pub fn create_proto<'gc>(
+    gc_context: MutationContext<'gc, '_>,
+    proto: Object<'gc>,
+    fn_proto: Object<'gc>,
+) -> Object<'gc> {
+    let mut object = ScriptObject::object(gc_context, Some(proto));
+
+    object.force_set_function(
+        "toString",
+        to_string,
+        gc_context,
+        DontDelete | ReadOnly | DontEnum,
+        Some(fn_proto),
+    );
+
+    object.into()
+}
+
+fn to_string<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let name = this.get("name", activation, context)?;
+    let name_str = name.coerce_to_string(activation, context)?;
+
+    let message = this.get("message", activation, context)?;
+    let message_str = message.coerce_to_string(activation, context)?;
+
+    let formatted_message = if message_str.is_empty() {
+        name_str.to_string()
+    } else {
+        format!("{}: {}", name_str, message_str)
+    };
+
+    Ok(formatted_message.into())
```

Comment:
Looks like it only prints out the `message` for me:

```
var e = new Error();
e.name = "name";
e.message = "message";

trace(e.toString()); // message
```

Reply:
Ah, I was following AS3 here as well. I'll fix it in that case